### PR TITLE
Add documentation of the data model

### LINKED ISSUE #8: Place some details about the data storage
(More) details on the data storage should go into the readme and manpage.

We already wrote some docs about our data storage as part of a project documentation, which we handed over to our university. It wouldn't be much work extracting the relevant parts. Maybe we should have done this _before_ people on the internet asked about it/read things up in the source code.

### PR DESCRIPTION
**Pull request by neithernut**
Resolves #8.

### CODE REVIEW

**Comment by neithernut** [comment]
I really made very little progress so far. This is partly because I have to write the documentation from scratch (because we wrote the external docs in german), partly because of other stuff coming up.

**Comment by matthiasbeyer** [comment]
When writing the documentation, please also read [the reddit thread](https://www.reddit.com/r/git/comments/5w55pp/gitdit_decentralized_issue_tracking_for_git/) and [the hackernews thread](https://news.ycombinator.com/item?id=13732598), to answer the basic questions asked there.

File: doc/datamodel.md [hunk 1]
```diff
@@ -6,3 +6,55 @@ in blobs within the tree. Similar to threads in a mailing list, issues and
...
+
+An issue is identified by the hash of its initial commit. For each issue, a
+"head" references `refs/dit/<issue-hash>/head` is created. Other message refer
+to the replied message via their first parent. Other parents may be used to
+refer to other commits, e.g. in bug-reports or cross-references to messages in
+other issues.
+
+The message trees are spanned via "leaf" references which are maintained in
+`refs/dit/<issue-hash>/leaves/` for each issue. Leaf references which are not
+required any more for preventing garbage collection of messages.
```
Comment (matthiasbeyer) [inline doc/datamodel.md]: This is not a sentence.

File: doc/datamodel.md [hunk 2]
```diff
@@ -6,3 +6,55 @@ in blobs within the tree. Similar to threads in a mailing list, issues and
...
+A maintainer may update the head reference to a specific point in the
+discussion. However, she may also choose to maintain an independent sequence of
+status changes, referring to messages in the discussion through its second
+parent. At this point, we do not yet provide tools for managing the head
+reference of an issue.
+
+Since the initial message of an issue can be identified by the presence of an
+associated head reference, it can safely refer to arbitrary commits as parents.
+For example, a bug report may refer to a specific commit in which a bug was
+observed. A patch set, oh the other hand, clearly should be rooted in the
```
Comment (matthiasbeyer) [inline doc/datamodel.md]: Typo `"oh"`.

File: doc/use-cases.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+repository other than the source code repository and still have issues visible
+via the remote in a developer's local repo.
+
+For example, bugs encountered in some programs are manifestations of bugs in
+libraries the program depends on. Currently, in such cases, a bug is often filed
+by hand in the dependency's bug tracker. With git-dit, the issue can be simply
+transferred or referenced in the dependency's (bug) repository.
+
+Similarly, an organization or company, for example, may use a single repository
+as a bug tracing database to which a certain set of customers (or even just a
```
Comment (matthiasbeyer) [inline doc/use-cases.md]: Typo here.

**Comment by neithernut** [comment]
I consider this done, for now.

File: doc/caveates.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+Caveats
+=======
+
+Git-dit is, obviously, git specific and will hence not work with other VCS like
+mercurial. Import and export functionality for other issue tracking systems are,
```
Comment (matthiasbeyer) [inline doc/caveates.md]: Wouldn't mention another VCS by name here. It does not work with others. Period.

File: doc/caveates.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+however, not inconceivable. Functionality like referencing issues across
+repositories will, however, most likely never be provided across different VCS.
```
Comment (matthiasbeyer) [inline doc/caveates.md]: Wouldn't mention that either.

**Comment by neithernut** [comment]
I will probably rebase this once #18 is merged.

**Comment by neithernut** [comment]
Mh... looking back this PR somewhat diverged from it's original goal.

**Comment by neithernut** [comment]
I also did not do the usual line-break-after-each-sentence thing, since markdown is often read in plain text.

File: doc/caveats.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+Caveats
+=======
+
+Git-dit is, obviously, git specific and will hence not work with other VCS like
+mercurial. Import and export functionality for other issue tracking systems are,
```
Comment (matthiasbeyer) [inline doc/caveats.md]: My annotation here was probably lost due to the rebase: Do not mention a VCS by name here. Just "others" is sufficient.

File: doc/caveats.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+however, not inconceivable. Functionality like referencing issues across
+repositories will, however, most likely never be provided across different VCS.
```
Comment (matthiasbeyer) [inline doc/caveats.md]: Wouldn't include this sentence. Do not tell people what won't be there; just tell them what _will_ be there.

Comment (neithernut) [inline doc/caveats.md]: If you remember some segments about "next generation" databases, one of the most prominent failures when adapting them was people not realizing functionality which they presumed would be there was missing.

**Pull request merged**